terra-money/finder
# Add TNS support

## Changes
- Add related TNS queries & functions to `src/libs/tns.ts`
- Add `isTnsName` check in `src/scripts/utility.ts`
- Add address resolution before rendering in `src/pages/Account/Address.tsx`

## Test cases
- Fill in `test-02.ust`, it should resolve to a smart contract page of address `terra1jfk9saed6c4n40qwe54mcxmgs8kz4husly3jx4`
- Fill in `blossom.ust`, it should resolve to an account page of address `terra1ftxft7kwvhw0c5m972rqg7gak9764wu9hthwzp`

## Example Behavior
![Kapture 2021-11-08 at 13 26 17](https://user-images.githubusercontent.com/93900505/140694207-4aa62cdd-cd07-48d3-b3a8-1dbba9fa56d7.gif)
![Kapture 2021-11-08 at 13 29 32](https://user-images.githubusercontent.com/93900505/140694497-d6d7073f-6ad6-4d23-8ef4-b50136f2efb8.gif)
